src/player.cpp

```diff
@@ -11489,8 +11489,15 @@ void player::read(int inventory_position)
         assign_activity( ACT_READ, time - moves, -1, inventory_position );
         // Never trigger studying when skimming the book.
         activity.values.push_back(0);
+        // pushes item_frm into 2nd element
+        activity.values.push_back(inv_from);
+        activity.values.push_back(item_fr_loc.x);
+        activity.values.push_back(item_fr_loc.y);
+        activity.values.push_back(item_fr_loc.z);
+        activity.values.push_back(item_fr_veh?-1:0);
+        activity.str_values.push_back(it->typeId());
```

Comment:
Cool man... Looks so beautiful...

Haha hope that 1 day I can reach your standard